app/controllers/dashboard/accounts_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Dashboard
+  class AccountsController < DashboardController
+    def settings
+      if !current_user.account.trial? && current_user.account.stripe_customer_id.blank?
+        flash[:alert] = flash_alert_text
+        return
+      end
+
+      return if current_user.account.stripe_customer_id.blank? || current_user.account.inactive?
```

Comment:
Yes, good idea moving the methods over. But the check is not duplicate, since it could be that the first check on line 6 passes but the `stripe_customer_id` is still blank.

Reply:
Thought about it again. I don't think adding alias to the User or Account model would vastly improve things here. I would just leave it.

Reply:
> (or alternatively in the Account model)
> Also, isn't it confusing to move things that are related to the Account to the User model?

Yeah, maybe better suited in the Account model directly.

I meant two methods to wrap these two conditions:
1/ `current_user.account.stripe_customer_id.blank? || current_user.account.inactive?`
and
2/ `!current_user.account.trial? && current_user.account.stripe_customer_id.blank?`

It would help readability

Reply:
Ok, got it!